metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/BasicStellarTest.java

```diff
@@ -916,6 +916,53 @@ public void testShortCircuit_conditional() throws Exception {
     Assert.assertEquals("foo", run("if NOT('metron' in [ 'metron', 'metronicus'] ) then THROW('expression') else 'foo'", new HashMap<>()));
   }
 
```

Comment:
These tests are good, but these evaluations should be tested with other things in the things in the if s

IF SOME_MORE_COMPLEX_FUNCTION THEN IF LAMBDA THEN MATCH ELSE B ELSE C
To make sure we are not messing the stack up.

Reply:
if you look at https://github.com/apache/metron/pull/814, which was when I did checkpoint work for match, I had a lot of fallout to account for.  Maybe you can take a peak at the match tests.

Reply:
yeah, that is the idea.  I had to add more and more permutations ( I added tests for issues as @jjmeyer0 found them, or as @cestella found them )
If you have one mega case that works that is fine.  I started out differently, because I had things that did not work at the beginning and I added later ( MAP support etc )